community/data-structure-on-rocksdb.md

```diff
@@ -265,6 +265,39 @@ key|version|EID MS|EID SEQ => |     encoded value     |
                               +-----------------------+
 ```
 
+#### stream consumer group metadata
+
+The consumer group metadata contains the basic information of the consumer group used in XINFO GROUPS. The key is composed by the stream key, version, group name and a metadata delimiter. The metadata delimiter is a const string `METADATA` in the implementation. The value is composed by consumer number, pending number, last delivered id, entries read and lag.
+
+```text
+                                   +-----------------+----------------+-------------------+--------------+-------+
+key|version|group_name|METADATA => | consumer number | pending number | last delivered id | entries read |  lag  |
+                                   |     (8byte)     |     (8byte)    |      (8byte)      |    (8byte)   |(8byte)|
+                                   +-----------------+----------------+-------------------+--------------+-------+
+```
+
+#### stream consumer metadata
+
+A consumer group contains several consumer and each consumer also has a metadata which is used in XINFO CONSUMERS. The key is composed by the stream key, version, group name, consumer name and a metadata delimiter. The metadata delimiter is a const string `METADATA` in the implementation. The value is composed by pending number, last idle time and last active time.
```

Comment:
=>

A consumer group contains several consumer and each consumer also has its own metadata which is used in the `XINFO CONSUMERS` command. 
The key starts with the stream key, version, consumer group name, consumer name. The next segment is a delimiter - a string `METADATA` (hardcoded value). 
After the delimiter, there are:
- the number of entries in the PEL
- the timestamp of the last consumer interaction
- the timestamp of the last successful consumer interaction (actually read or claimed some entries).